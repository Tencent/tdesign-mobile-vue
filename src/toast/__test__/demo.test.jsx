/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import iconTextVue from '@/toast/demos/iconText.vue';
import maskVue from '@/toast/demos/mask.vue';
import mobileVue from '@/toast/demos/mobile.vue';
import overlayVue from '@/toast/demos/overlay.vue';
import positionVue from '@/toast/demos/position.vue';
import preventScrollThroughVue from '@/toast/demos/preventScrollThrough.vue';
import textVue from '@/toast/demos/text.vue';

const mapper = {
  iconTextVue,
  maskVue,
  mobileVue,
  overlayVue,
  positionVue,
  preventScrollThroughVue,
  textVue,
};

describe('Toast', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Toast ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
