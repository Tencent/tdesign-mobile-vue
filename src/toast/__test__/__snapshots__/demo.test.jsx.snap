// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toast > Toast baseVue demo works fine 1`] = `
<div
  class="toast-demo"
>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      纯文本
      
    </span>
    <!---->
  </button>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      多行文字
      
    </span>
    <!---->
  </button>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      带横向图标
      
    </span>
    <!---->
  </button>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      带竖向图标
      
    </span>
    <!---->
  </button>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      加载状态
      
    </span>
    <!---->
  </button>
</div>
`;

exports[`Toast > Toast closeVue demo works fine 1`] = `
<div
  class="toast-demo"
>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      显示提示
      
    </span>
    <!---->
  </button>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      关闭提示
      
    </span>
    <!---->
  </button>
</div>
`;

exports[`Toast > Toast coverVue demo works fine 1`] = `
<div
  class="toast-demo"
>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      禁止滑动和点击
      
    </span>
    <!---->
  </button>
</div>
`;

exports[`Toast > Toast mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <h1
    class="title"
  >
    Toast 轻提示
  </h1>
  <p
    class="summary"
  >
    一种轻量级反馈或提示，不会打断用户操作。
  </p>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        01 组件类型
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        基础提示
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="toast-demo"
      >
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            纯文本
            
          </span>
          <!---->
        </button>
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            多行文字
            
          </span>
          <!---->
        </button>
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            带横向图标
            
          </span>
          <!---->
        </button>
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            带竖向图标
            
          </span>
          <!---->
        </button>
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            加载状态
            
          </span>
          <!---->
        </button>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        01 组件状态
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        内置主题
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="toast-demo"
      >
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            成功提示
            
          </span>
          <!---->
        </button>
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            警告提示
            
          </span>
          <!---->
        </button>
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            错误提示
            
          </span>
          <!---->
        </button>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        03 显示遮罩
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        弹窗可显示遮罩，禁止滑动和点击
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="toast-demo"
      >
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            禁止滑动和点击
            
          </span>
          <!---->
        </button>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        04 手动关闭
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        手动关闭轻提示
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="toast-demo"
      >
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            显示提示
            
          </span>
          <!---->
        </button>
        <button
          aria-disabled="false"
          class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
          role="button"
          type="button"
        >
          <!---->
          <span
            class="t-button__content"
          >
            
            关闭提示
            
          </span>
          <!---->
        </button>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`Toast > Toast themeVue demo works fine 1`] = `
<div
  class="toast-demo"
>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      成功提示
      
    </span>
    <!---->
  </button>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      警告提示
      
    </span>
    <!---->
  </button>
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      错误提示
      
    </span>
    <!---->
  </button>
</div>
`;
